### PR TITLE
Http auth

### PR DESCRIPTION
**Pull request by oleksiishyman**
Implements http endpoint for auth using keystone token

### CODE REVIEW

File: pkg/http_endpoint/authentication/jwt_authentication.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+)
+
+const contextJWTProperty = "AuthToken"
+
+// JWTIssuer
+const tokenIssuer = "visualization-api"
+
+// CustomClaims defines what data would be stored in jwt token
+type CustomClaims struct {
+	IsAdmin   bool   `json:"is_admin"`
```
Comment (funrun11) [inline pkg/http_endpoint/authentication/jwt_authentication.go]: Please use isAdmin according to spec http://petstore.swagger.io/?url=https://raw.githubusercontent.com/zaletniy/visualization-api/master/doc/visualization-api-1.0.0.yaml

File: pkg/http_endpoint/authentication/jwt_authentication.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+// JWTTokenFromParams creates jwt token given claims data
+func JWTTokenFromParams(secret string, isAdmin bool, projectID string,
+	expiresAt time.Time) (string, error) {
+
+	claims := CustomClaims{
+		isAdmin,
+		projectID,
+		jwt.StandardClaims{
+			ExpiresAt: expiresAt.Unix(),
```
Comment (funrun11) [inline pkg/http_endpoint/authentication/jwt_authentication.go:29]: Why not to use RFC3339 with UTC timezone for better readability?

File: pkg/http_endpoint/v1/handler.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package v1Api
```
Comment (funrun11) [inline pkg/http_endpoint/v1/handler.go:1]: let us use underscore as we have it already

File: pkg/http_endpoint/v1/router.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"visualization-api/pkg/logging"
+)
+
+const adminAPIPrefix = "/admin"
+const authPrefix = "/auth"
+
+func authRouter(clients *common.ClientContainer,
+	handler common.HandlerInterface, secret string) *chi.Mux {
+	router := chi.NewRouter()
+	router.Get("/openstack", func(w http.ResponseWriter, r *http.Request) {
```
Comment (funrun11) [inline pkg/http_endpoint/v1/router.go]: We agreed that it is a POST

File: pkg/http_endpoint/authentication/jwt_authentication.go (hunk 1 above)
Comment (oleksiishyman) [reply pkg/http_endpoint/authentication/jwt_authentication.go]: This data is encoded inside JWT token
Real http response json exactly matches spec and has "isAdmin" field

I would change it to isAdmin not to confuse users with 2 different values

File: pkg/http_endpoint/authentication/jwt_authentication.go (hunk 2 above)
Comment (oleksiishyman) [reply pkg/http_endpoint/authentication/jwt_authentication.go:29]: This data is encoded inside JWT token and is not visible to use, so I've chosen shortest format possible.
Real http response has format, that corresponds to spec

File: pkg/http_endpoint/v1/handler.go (hunk 3 above)
Comment (oleksiishyman) [reply pkg/http_endpoint/v1/handler.go:1]: This is not allowed by linters

File: pkg/http_endpoint/authentication/jwt_authentication.go (hunk 2 above)
Comment (funrun11) [reply pkg/http_endpoint/authentication/jwt_authentication.go:29]: IntDate  A JSON numeric value representing the number of seconds from
      1970-01-01T0:0:0Z UTC until the specified UTC date/time.  See RFC
      3339 [RFC3339] for details regarding date/times in general and UTC
      in particular.

Is required format for date values according to https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-15#section-4.1.4

**Comment by funrun11** [comment]
@oleksiishyman please also add mock-generate goal to Makefile in order to be able to regenerate all mocks

File: pkg/http_endpoint/v1/handler.go (hunk 3 above)
Comment (funrun11) [reply pkg/http_endpoint/v1/handler.go:1]: Ok.

File: pkg/http_endpoint/authentication/jwt_authentication.go (hunk 2 above)
Comment (funrun11) [reply pkg/http_endpoint/authentication/jwt_authentication.go:29]: Fine then.

File: pkg/http_endpoint/authentication/jwt_authentication.go (hunk 1 above)
Comment (funrun11) [reply pkg/http_endpoint/authentication/jwt_authentication.go]: According to our agreement let us keep JWT token format separate from http response, so we could minimize token size.

**Pull request merged**